### ISSUE TITLE
Does an Offer always have a planned end date?

### ISSUE THREAD

**Issue by rjyounes** [semanticarts/gist#627]
Current definition of `gist:Offer`:

```
gist:Offer
	a owl:Class ;
	owl:equivalentClass [
		a owl:Class ;
		owl:intersectionOf (
			gist:ContingentObligation
			[
				a owl:Restriction ;
				owl:onProperty gist:hasDirectPart ;
				owl:someValuesFrom gist:CatalogItem ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:hasGiver ;
				owl:someValuesFrom [
					a owl:Class ;
					owl:unionOf (
						gist:Organization
						gist:Person
					) ;
				] ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:hasMagnitude ;
				owl:someValuesFrom gist:Monetary ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:plannedStartDateTime ;
				owl:cardinality "1"^^xsd:nonNegativeInteger ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:plannedEndDateTime ;
				owl:cardinality "1"^^xsd:nonNegativeInteger ;
			]
		) ;
	] ;
	skos:definition "A commitment to buy or sell a described or identified part or service."^^xsd:string ;
	skos:prefLabel "Offer"^^xsd:string ;
	.
```

**Comment by dylan-sa** [comment]
It might be right that all offers lapse at some point (even if only after a reasonable period of time), but I wouldn't think it has to be planned for. I see that the latest version of the formal definition for `gist:Offer` changes `gist:plannedStartTime` to `gist:hasStart`. One option is to do the same for the end time. Another option is to remove the end time restriction. I think I would lean towards the latter. Even if, as a matter of law, all offers lapse at some point, having an end time doesn't seem to be part of what it means for something to be an offer.

**Comment by rjyounes** [comment]
The version above is the not-yet-released gist 11. gist 10, as you note has restrictions on `hasStart` and `hasPlannedEnd`. I'm not sure if the change from planned or actual start to planned start was deliberate or an accident, but it seems like an improvement.

**Comment by uscholdm** [comment]
I agree that its wrong to say every Offer has a planned end. Jus tremove that restriciont, is the way to go.

**Comment by uscholdm** [comment]
@rjyounes, @dylan-sa 

> I see that the latest version of the formal definition for gist:Offer changes gist:plannedStartTime to gist:hasStart.

That is no longer the case, it is now `plannedStartDateTime`. 

PROPOSAL: make 3 changes
1. remove restriction for end time
2. use startDateTime for the start restriction, (be agnostic on planned vs. actual)
3. Fix a bug in the definition, it should say _product_, not _part_

New version would be this:
```
gist:Offer
	a owl:Class ;
	owl:equivalentClass [
		a owl:Class ;
		owl:intersectionOf (
			gist:ContingentObligation
			[
				a owl:Restriction ;
				owl:onProperty gist:hasDirectPart ;
				owl:someValuesFrom gist:CatalogItem ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:hasGiver ;
				owl:someValuesFrom [
					a owl:Class ;
					owl:unionOf (
						gist:Organization
						gist:Person
					) ;
				] ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:hasMagnitude ;
				owl:someValuesFrom gist:Monetary ;
			]
			[
				a owl:Restriction ;
				owl:onProperty gist:startDateTime ;
				owl:cardinality "1"^^xsd:nonNegativeInteger ;
			]
		) ;
	] ;
	skos:definition "A commitment to buy or sell a described or identified product or service."^^xsd:string ;
	skos:prefLabel "Offer"^^xsd:string ;
	.
```

**Comment by rjyounes** [comment]
I agree with @uscholdm's proposal. Should we change `hasDirectPart` to `hasPart`?

**Comment by uscholdm** [comment]
> Should we change hasDirectPart to hasPart, just to hedge our bets?

I don't think so. An offer should point directly to thing being offered.

**Comment by dylan-sa** [comment]
I like @uscholdm's proposal.

**Comment by rjyounes** [comment]
> I don't think so. An offer should point directly to thing being offered.

Now that you put it like that, I don't think `has(Direct)Part` is the right predicate. The thing being offered has a more specific connection to the offer than part-whole. I've [proposed the introduction of a predicate `isOfferOf`](https://github.com/semanticarts/gist/issues/528). It doesn't have to be that, but "part of" just doesn't seem like the right semantic relationship.

**Comment by uscholdm** [comment]
@rjyounes 
> Now that you put it like that, I don't think has(Direct)Part is the right predicate. The thing being offered has a more specific connection to the offer than part-whole. I've https://github.com/semanticarts/gist/issues/528. It doesn't have to be that, but "part of" just doesn't seem like the right semantic relationship.

We don't want to introduce a new predicate unless the current one is clearly wrong or ambiguous.  Its general, but not wrong. I cannot think how ambiguity would arise.  What other meaning could there possibly be to a `directPart` link to a `CatalogItem` other than being that which is offered?
